ai/openai/v0/client.go

```diff
@@ -1,48 +1,21 @@
 package openai
 
 import (
-	"github.com/instill-ai/component/internal/util/httpclient"
 	"go.uber.org/zap"
 	"google.golang.org/protobuf/types/known/structpb"
-)
 
-func newClient(setup *structpb.Struct, logger *zap.Logger) *httpclient.Client {
-	c := httpclient.New("OpenAI", getBasePath(setup),
-		httpclient.WithLogger(logger),
-		httpclient.WithEndUserError(new(errBody)),
-	)
+	openaiclient "github.com/sashabaranov/go-openai"
+)
 
-	c.SetAuthToken(getAPIKey(setup))
+func newClient(setup *structpb.Struct, logger *zap.Logger) *openaiclient.Client {
 
+	cfg := openaiclient.DefaultConfig(getAPIKey(setup))
 	org := getOrg(setup)
 	if org != "" {
-		c.SetHeader("OpenAI-Organization", org)
+		cfg.OrgID = org
 	}
 
-	return c
-}
-
-type errBody struct {
-	Error struct {
-		Message string `json:"message"`
-	} `json:"error"`
-}
-
-func (e errBody) Message() string {
-	return e.Error.Message
-}
-
-// getBasePath returns OpenAI's API URL. This configuration param allows us to
-// override the API the connector will point to. It isn't meant to be exposed
-// to users. Rather, it can serve to test the logic against a fake server.
-// TODO instead of having the API value hardcoded in the codebase, it should be
-// read from a setup file or environment variable.
-func getBasePath(setup *structpb.Struct) string {
```

Comment:
This is still relevant for the tests (when introduced), we can use the [`NewClientWithConfig`](https://github.com/sashabaranov/go-openai/blob/master/client.go#L55) method to test the calls against a test server

Reply:
It turns out that `go-openai` has limited support for optional parameters, so I’m going to revert this PR.